File: Jack_Charris/Programas/Pruebas/PDart/Promedio.dart
import 'dart:io';
// nos permite interactuar con la consola (las estiquetas de dart)

void main() {
    print('Digita los numeros separados por espacios: ');
    String input = stdin.readLineSync()!;
    // nos permite poder escribir en la consola

    List<String> numerosComoString = input.split(' ');
    // divide la cadena en subcadenas cada vez que encuentra un espacio. Esto nos da una lista de números en formato de texto.
    List<double> numeros = numerosComoString.map(double.parse).toList();
    // Usamos map() para convertir cada elemento de la lista de strings a un número de punto flotante (double). double.parse convierte un string en un número de punto flotante. Luego, llamamos a toList() para convertir el iterable resultante en una lista.


    double suma = 0;
    for (var numero in numeros){
        suma += numero;
    }
    // Recorremos la lista de números y sumamos cada número a la variable suma.

    double promedio = suma / numeros.length;
    // saca el promedio de los numeros sumados
    // La propiedad length se utiliza en Dart para obtener el número de elementos en una lista.

    print('El promedio de los numeros ingresados es: $promedio');
    // imprimimos el resultado.

}

File: Jack_Charris/Programas/Pruebas/PRuby/promedio.rb
# Nos permite interactuar con la consola

puts 'Digita los numeros separados por espacios: '
input = gets.chomp
# Nos permite leer una línea de entrada desde la consola

numeros_como_string = input.split(' ')
# Divide la cadena en subcadenas cada vez que encuentra un espacio. Esto nos da una lista de números en formato de texto.

numeros = numeros_como_string.map(&:to_f)
# Usamos map() para convertir cada elemento de la lista de strings a un número de punto flotante (float). &:to_f es una forma abreviada de escribir { |s| s.to_f } en Ruby.

suma = numeros.reduce(0, :+)
# Recorremos la lista de números y sumamos cada número a la variable suma. reduce(0, :+) suma todos los elementos de la lista, comenzando desde 0.

promedio = suma / numeros.length
# Calcula el promedio de los números sumados. La propiedad length se utiliza en Ruby para obtener el número de elementos en una lista.

puts "El promedio de los numeros ingresados es: #{promedio}"
# Imprimimos el resultado.